### ISSUE TITLE
I would like to pick your brain for a minute. 

### ISSUE THREAD

**Issue by kdschlosser** [Joshua-Data-Wizard/PyRealEstate#10]
I have a question for ya. This is regarding anomalies seen when making a PRD graph. 

There are some 17,000 properties being used to create this PRD. So here is the setup.

The county has a time adjustment factor of 1.5% per month. this is calculated using the last assessment value of the property divided by the sale price and then ran through a Multiple Linear Regression model per the county's documentation. I used the counties 1.5% in the PRD and the graph has this going on when I zoom into it.

![image](https://github.com/Joshua-Data-Wizard/PyRealEstate/assets/10932297/31f1cc23-c162-4941-b710-61abe565073a)

Now When I run the model outlined by the county to get the time trending factors they do not come out to 1.5% per month. It comes out to more like 2.2% per month. The county has the properties grouped into market areas and each market area has it's own time trend. The county states that all market areas across the county are all 1.5%. I find that rather odd that this has occurred I mean mathematically the chances of it happening are so small that it could be considered that it would never happen. But whatever, I'll go with it. You can see the vertical banding of the ratios. I believe this is being caused by the time trending factors the county has calculated (or pulled out of a hat).

Here is the graph when using the time trending calculation I generated

![image](https://github.com/Joshua-Data-Wizard/PyRealEstate/assets/10932297/b4d066d0-3d2c-4e93-87cf-e90e688f7de9)

You can still see faintly some vertical banding. It is nothing like what is seen when using the counties time trend. I believe this faint vertical banding is being caused by fudged time trending factors being used in a previous assessment year. Since the time trends use the prior assessments to calculate the time trend if there was banding in the data then it would continue to be seen in the data for a very long time. The banding would get amplified if the county does what they did in a prior assessment year correct? And that is the reason why it is extremely defined when using the counties time trends.

I am finding some really crazy things in the data from the county. It's things like this graph that shouldn't exist. 

Is my understanding if what is happening consistent with what you see in the graphs? Is this being caused by bogus time trend numbers the county is using?

**Comment by kdschlosser** [comment]
Here is an image of the complete graph using the counties time trending factors.

I have added an MLR line to this graph to plot the regression in values across the time adjusted sales prices. You can see there is a considerable slope to the line as the property prices increase the properties are being undervalued is increasing in the amount they are being undervalued. and the ratio of undervalued properties to over over valued for a given time adjusted sale amount range is also increasing.

![image](https://github.com/Joshua-Data-Wizard/PyRealEstate/assets/10932297/694b96a2-4c8d-419a-90c3-f33e07980343)


Does this graph look right? It doesn't look right to me. It is almost the same thing that is seen in a 3rd party analysis that the state had done and the state has signed off on the analysis stating it is within norms.

It just doesn't look right to me. There should be an even disbursement above and below the 1.0 line across the entire time adjusted sale price range. Not an increasing amount of number of over values the lesser the price of the property. and the opposite of that the more expensive the property.

**Comment by Joshua-Data-Wizard** [comment]
Good Morning Kevin, I hope that all is going well with you! So I will do my best to answer the question but please if anything doesn't make sense or you would like me to go into further detail please do not hesitate! 

I agree that the chances that all the market areas have the same exact time trend is fairly rare. One reason that this might be is that an assessor office needs to be able to explain their adjustments to all taxpayers and sometimes arguing a different area has a different rate can be slightly tumultuous so one solution is to have a county / jurisdiction time trend and then have location adjustments for the market areas and say that everything in the jurisdiction moved by 1.5% but since this market area has become less desirable we have applied some negative factor and this market area has become very desirable so it has a positive factor. By doing this it would account for slight differences in the time trend rates by area.  In addition, some assessors will include the raw building cost of properties in the model and if they have increased the raw cost due to increase in materials this might be why the time trend rate is slightly lower than expected.  Side note when looking at some jurisdictions such as the ones in Florida by the water their time trends are really interesting! You can see seasonal adjustments with sales in the winter being slightly elevated compared to summer, which is where the nonlinear adjustment curve calculations in the package came from :)  I do want to put emphasis on everything that I just said is speculative and I am truly not sure if that is the case here. 


So depending on how the time trend itself is being used this potentially could be the case. The big problem with using percent change is the low stuff will stay low and the high stuff will stay high for the lack of better terms :) However if you are building a new regression model if the time trend is slightly off or causing this banding then updating the coefficients of other variables could account for this. I have seen time trending used in a couple different manners. 

1. Just applying a time trend to the value - this leads to the problem above if the last year value was incorrect by using a percent change the issue will slowly compound as you pointed out. 

2. Calculate the average sale price per square foot (or some other unit of price) over time and derive a time trend and once all the sales have been time trended use a regression model to determine the impact on value for all other features - this is what the time trending libraries are more or less aimed at. This should relieve that compounding issue because the time trend is only derived from sales and then training a regression model to predict value on the sales adjusted by the time trend derived from sales should lead to equitable values for all.

3. When building your model directly incorporate time - this is one of my preferred methods; simply have the outcome of your regression model be the sale price and one of the predictors be time ( months since January 2019 for example). This will allow the model to adjust for confounding with other variables and then you could extract the time trend right out of the model (hopefully something to add to this library). 

Also some counties when referring to creating a regression model might be manually controlling some to all coefficients instead of using the regression algorithms to  determine them. Again not sure which is going on here just trying to be informative and give you all possible options! 


So to answer the last part this might be a little long winded so please don't get too upset! So an assessor's office always has the burden of having to be able to defend their value to a taxpayer and many jurisdictions have VAB's (value adjustment boards) or something similar where a taxpayer can dispute their value. The reason I am saying this is because unfortunately many times with very high dollar parcels the sales are slightly more disparate; this causes two issues one meaning from an appraisal standpoint they do not have as many comparables to support a higher value and unfortunately depending on the situation this can really force the hand of an assessor to come in lower than what a parcel might sell for when dealing with high dollar parcels.  The second issue is from a modeling standpoint. A good example of this is when thinking again in Florida for example if you were using the distance to open water as a predictor for houses the relationship of distance to open water and a houses value is not linear at all and many times the number of square feet of a house to sale price is also not linear. This is an issue because if you are using just linear regression it cant handle non linear relationships and since it's trying to increase the R squared from the models standpoint it says hey if I get the average parcels right who cares about the outliers I am pretty close overall and this can lead to specifically what you are seeing with low value parcels being over assessed and high values being under... So how do you potentially fix this? I am glad you asked 

1. You could switch to using multiplicative regression in jurisdictions that are diverse in properties; this can work well but the drawbacks is it does not completely account for it and in many cases of nonlinearities with multiple magnitude and direction changes this will not work and since you are using multiplicative regression the coefficients no longer mean for one unit increase the price increases this much. This can make the model harder to understand and explain.  

2. Implement splicing to variables - by this I mean for example with square footage of a house break it into groups  0-1000, 1000-3000, 3000-5000, 5000+ . By breaking it into groups you can try to get a linear line to explain this section of data . The issue with this is how many groups and where to break them. There are some statistical ways to do this :) 

3. Implement Machine Learning Models- This has become my favorite. Algorithms such as Light Gradient Boosting , XG Boost, Cat Boost can capture the non linearities and give very high levels of accuracy to address an inequality and handle the toughest data. The down side is someone needs to really know how to do hyperparameter tuning on these to reach optimal performance and then to understand how the model made its predictions and extract adjustments you need to use a model explainability algorithm such as LIME or SHAP.

Truly with all my responses to the problem is you need to start implementing high levels of technology and many jurisdictions have not reached this level yet. Also, it is key to note that my favorite machine learning is extremely early to property appraisal and not very accepted yet which can cause concern for many assessors in that right led alone. 

When looking at that graph I would say some of the really high ratios close to zero could probably be excluded after reviewing the sales. Many times those might be family members selling to one another for considerably lower amounts, it could also be a distressed sale (someone getting foreclosed on or passing away unfortunately). Also the number of sales above around 3.5 million seem to be fairly small. This  really leads into a sample size issue, you do not want "chase sales" for example if you have 10 sales of 3 million to 5 million dollars parcels but you have 300 of them in your jurisdiction do you want the 10 to determine the value of all of them?  This leads into being able to support the value from the assessors stand point.


All this being said I do not disagree with you one bit and your conclusion is right, ideally it should be all centered around one over the entire price range this is truly the equity everyone is searching for!  Unfortunately, due to some of the situations I mentioned above this is truly not that unusual to see and almost I would say more or less typical. Compared to many other jurisdictions I have seen this really isn't bad at all though :)  Going back to the qualified sales I loosely mentioned before if you are using all sales and not what are just qualified sales where there are no other influence (related buyers, distressed seller, the parcel being sold includes items from the property, commercial buildings are being sold with a tenant in place) this could skew your results. I really hope this helps and answers your question! please if you want me to go over anything please don't hesitate to ask I would be more than happy to!

**Comment by kdschlosser** [comment]
First Thing is the county I live in is 774 square miles in size. To put that into perspective, Rhode Island is 1,214 sqmi. There are approximately 220k residential properties in the county with about 14k sales a year.

How they state the time model is done is they take the selling price and divide it by the last assessment value they feed that along with the number of months from the last assessment to the selling date. That a linear regression is done. They do that for all properties that have sold during the 2 years following the last assessment.

Our county is broken up into 9 different market areas. the records for the time trends go back to 2009 and for every single assessment year except the current one you see different time trend factors based on the market area and also the improvement type. condos, single family, duplex/triples, townhomes and vacant land. For every assessment year these are all different and they are also different between market areas. This year everything is set at 1.5 except for condos an vacant land. Vacant land is at 1.05 and condos are at 1.60. Those numbers is the percentage increase per month that has to occur from the selling date to the end of the 2 year time frame.

Once that is done then we get into making adjustments for improvement size, heating type, extra buildings, decks, bathrooms, land contributors like view and exposure, neighborhood... I know there are more I would have to go and look. They have a deck slated at 41.00 dollars a sqft for an average quality deck. A enclosed porch (which is a deck that has a roof and screen room) is at 42.00 a sqft for average quality. They have a garden level basement that is finished at 40 a sqft for average quality and a normal basement is set at 53 a sqft. A garden level is not fully underground, it has normal sized windows. How in the world is that at a lesser cost??

Some of this stuff is simply crazy what they have the prices set to. 


And that banding that is seen is not when using my time trending model. That's seen when using the 1.5 percent per month that the county states. The real time trending is over 2.2 percent per month and when it is used to help to alleviate a large chunk of the undervaluing that is taking place.

**Comment by kdschlosser** [comment]
Oh and the counties yearly budget is just a smidge over 1 billion. and the payroll is almost 300 million of that and they only work 4 days a week. 3200 total employees. average pay is 93K a year for1600 work hours. 


Almost makes a person that does your line of work almost wanna move here... 130k is the average pay for an assessor in the county I live in.

**Comment by Joshua-Data-Wizard** [comment]
While the 130K average salary would be really nice I think the thing that interests me the most is the 4 day work week lol . While there are advantages to having assessors office's be decentralized one of the slight drawbacks is that there are a great number of ways in which offices determine value; some are more sophisticated than others and employ more mathematical and statical logic than others. The reason we time trend all sales is so we can build a model (or build a model with time directly built into it) and include all the features like docks, decks, living area and the model will then be able to tell us what the adjustment should be by using statistics and thus the value is determined by the market. I won't say this works perfect 100% of the time but seems to be a very high performing way of determining value. So what you have described is why I created this library, I am not saying the methods created in this library are the absolute best but if someone wanted to incorporate machine learning or new models for mass appraisal they can take advantage of some of the work already done. I wanted to make the work available for everyone to capitalize on. I am hoping to have more functionality and actual models to create and help predict property values added sooner than later, a couple of projects have just kept me busy. But I promise to keep adding to the library as soon as I can!

**Comment by kdschlosser** [comment]
what I find really screwy is that the county has a home that sold for say 300K. The time adjusted sale price ends up at 400K. The county places a value of 325K on the home. Now when a person disputes their property tax they have to use the time adjusted value of the home. The county didn't even think the property was worth  the time adjusted value so how is it to be expected that the time adjusted value is to be used when using that property as a comparable? 

My county has actually set values on properties that have sold that is lower than the selling price, not just the time adjusted selling price. The value of that home should be locked in position at the time adjusted selling price plain and simple. That is how much someone paid for it then that is what the value of it should be. This happens all the time with  8-9 million dollar homes and they have county values of 1.5 and 2 million. Even Zillow has an estimated value of 8-9 million. The funny thing is if you look at properties like this on Zillow at the zestimate history you can see hen the county has done an assessment. The zestimate will show a several million drop in value over a single day. It stays low for 2 maybe 3 months and then it jumps up to where it should be in a single jump. The reason why it jumps up is because another home sold that get used as a comparable and that selling price resets the value of the property to where it should be on zillow , The property is still being taxed way under what it should be and it will  continue for 2 years when the assessment happens and the property only goes up 200K or 300K in value. 

There are multi million dollar properties that have had their values go down when the entire market went up by 38%. They were already undervalued but for some reason they went down even more.

**Comment by Joshua-Data-Wizard** [comment]
Got it, yeah unfortunately it does sound like there are some unusual value movements for sure. I am sorry to hear that this is happening to you. are you a software developer? I know where I am at we are always looking to get additional help with advancing Mass Appraisal if your inserted !

**Comment by kdschlosser** [comment]
I do a lot of different things. Mostly I work on my house, It would have been easier to tear it all down and start from scratch I think. It's a never ending processes it seems. what started as replacing a couple of deck boards turned into a complete gutting of the house down to the studs. One room at a time (for the most part).

I have always had an aptitude for computer/electronics related things but have not been afraid to get my hands dirty either. From construction to PCB design to Automotive repair/mechanics to programming I have knowledge about a lot of different things. 

I have a decent grasp at how the assessment process works and the math behind it. I mean I was able to reverse engineer the property search website for the county I live in so I could download all of the information about every single property in the county that was publicly available and build a database with all of that information. Then write the software needed to access that data to be able to train different models to come up with the value of a property that I could check against what the county has set for a value. Took about 2-3 weeks to put that all together. It's not complete and I don't have a user interface or anything like that but it does work and prints out the numbers for me or I can plot the information to a graph.

Never really thought about getting into anything dealing with assessment of properties. I am 50 and I have no prior experience in the field, I wouldn't think myself to be qualified especially since I don't have any formal education in the field either.

**Comment by Joshua-Data-Wizard** [comment]
Wow that is really impressive  the large range of things that you do, that is more than most including me lol.  It also explains your profile picture with working on your house.  Being able to gather all that information and make the models is remarkable especially in such a short time frame! Being able to write the software to extract the data store it and create models covers such a wide range of task, you have my admiration on being able to get all that done!  So the chief deputy for the jurisdiction I work for always says no one gets into appraisal and intends to stay. I really think he is write because when I started I didn't think I would be doing this permanently but at least for me I really grew fond of the challenge of trying to build better and newer models to be more fair and equitable. We have worked on different types or recursive feature elimination, geographically weighted regression and it keeps it pretty interesting! While you don't have any direct prior experience or  formal education you really seem to be off to a great start to me and know more than some that have been in the field for a while!

**Comment by kdschlosser** [comment]
If you go into my GitHub profile and on the bottom left you will see a section called achievements. check out the last 2 that are there. just hover the mouse over them.

I tell people all the time I really only know how to do one thing and that single thing allows me to do  everything else. I know how to read. I am not afraid to try. if it doesn't work most people consider that a failure. It's not so long as I have taken something away from it. An education is never a failure. It is never wasted time so long as something was learned. My profile picture is of me standing in a hole and my arm is resting on the sidewalk that leads to my front door. I had just finished removing a stump of a tree that had soaked up so much water that i was only able to get about 1/2" into it before it completely killed the teeth on my chainsaw. So I went at it the old fashion way, a shovel and an axe.

And what I know is all from reading what is on the internet. I never read a single thing and take it as fact I have to read information from several sources to make sure that I am understanding it correctly.

I am also a self taught programmer. 7 years ago the only thing I knew how to program was a VCR. I had started to do the automation thing to my home and I wanted the ability to control my gas fireplaces. Nothing existed and that is what got me into programming. I wrote the firmware for a micro controller that turned a simple relay on and off. I added a network adapter to it so it could be told what to do from my computer. As the say the rest is history.


Here is an example of the type of work  that I have done on my house.

![IMG_20230507_151953](https://github.com/Joshua-Data-Wizard/PyRealEstate/assets/10932297/657c5b03-8bcb-4e0c-9490-95f5ba499096)

![IMG_20230518_170954](https://github.com/Joshua-Data-Wizard/PyRealEstate/assets/10932297/567f3a38-3667-409d-bb1d-278e8d1712a6)

and here is a project vehicle of mine.
![IMG_20221002_211427](https://github.com/Joshua-Data-Wizard/PyRealEstate/assets/10932297/660b505e-efdf-4a22-8c30-348c58f4c3f3)

![IMG_20220921_222451](https://github.com/Joshua-Data-Wizard/PyRealEstate/assets/10932297/7df237f1-62de-4bd5-8912-a11ea20c5fd6)

![IMG_20220915_213740](https://github.com/Joshua-Data-Wizard/PyRealEstate/assets/10932297/8d99b3c5-dc34-492a-8588-53f16ba2e1d7)

**Comment by Joshua-Data-Wizard** [comment]
Those badges are really amazing, congratulations on them, they are beyond cool! That is a inspiring way of getting into programming! Your house looks like it is turning out really nice, it must be really cool getting to see the efforts of your work in such a tangible way! That is a Pontiac Solsitice correct? They are an awesome car!

**Comment by kdschlosser** [comment]
That is a Pontiac Solstice. That example has 50K miles on it. I have dome some work to the interior by wrapping all of the plastic parts inside with marine grade vinyl fabric. The interior was 100% black which took a small interior and made it feel even smaller. The red looks good with the yellow paint job and it makes the interior feel roomier while getting rid of the cheap GM plastic feel.

2800 lb car with an engine that puts out 500 HP and similar torque. Makes for a fun ride. I have done 100% of the work on that vehicle myself. from what you see done under the hood to the interior, it was all done by me. This coming spring I am going to paint the car. Never done that before so it should be a fun challenge.


All of the hoses under the hood for things like the vacuum lines and the coolant system are all military spec rated to 3500PSI burst pressure. If you notice there are no hose clamps on anything. It's all compression fittings. I assembled all of the hoses myself. That silver tube looking thing in the bottom of the middle photo I made from a piece of solid aluminum using a belt sander, cordless drill and metal files. Wanna know how long to takes to drill through a piece of 2" diameter x 9" long piece of aluminum using a 1 5/16" drill bit with an 18V cordless drill? it's not fast I will tell you that. I think it was 33 batteries worth.

The house is a love hate relationship. It keeps me in good physical shape. I live at 8500 foot elevation so it takes twice as much effort to accomplish the same task as where you are. I live in the sticks. just far enough away to not have to deal with people but close enough to have things like broadband internet. I have close to an acre of land so my neighbors are far enough away where you cannot see into their windows and they cannot see into mine.

**Comment by kdschlosser** [comment]
This is the goal. this is where I am trying to get to with the house.

https://www.youtube.com/watch?v=NnfQbKcRmNs

and this really goes into how much work I put into that 3D model of the house.

https://www.youtube.com/watch?v=aS5GQXNStQc

you have to excuse the low resolution and it going blurry from time to time. Not sure why that was happening.

**Comment by Joshua-Data-Wizard** [comment]
Wow that is really impressive I really like all the detail you put into the model , seeing the framing was going above and beyond! It is clear you put a lot of planning into this! I think one of my favorites was the movie theater on the lower level. This is really cool!

**Comment by kdschlosser** [comment]
The movie theater is already built. Well most of it. Gotta get the theater seats brass railings and flooring. other than that it's all finished  including the rack in the back of the room.

**Comment by Joshua-Data-Wizard** [comment]
That is so cool and I just noticed your comment about the Solstice. I did notice all the compression fittings and 33 batteries that is crazzzzyyyyy. I really appreciate all the detail and effort you put into all of your goals and tasks it really shows!

**Comment by kdschlosser** [comment]
I am a big fan of putting in 100% all the time. If someone is not going to put in 100% then don't bother doing it because it will end up having to get redone.

**Comment by kdschlosser** [comment]
OK so get this... last week one of the "managers" of the Division of Property Taxation send me a report from the State Forest Service. I had requested this report from the guy. He in turn went and asked the County Assessor's office for a copy of the report. I have not had any luck getting the county to respond to any of my requests for information and since I was already in conversation with the state I thought I would ask them to see if they can get it for me. He got a copy of the report for me and sends it to me in an e-mail. What was sent to me was an Excel file and what was in that file was property identification numbers and lot sizes for those properties. The report is supposed to be all of the properties in the county that have been zoned as "Forest Land" and information about  the annual inspections done on each property.  You would think that kind of report would be a PDF or something similar, not a spreadsheet. No heading or information that states the document came from the Forest Service no way to identify the origin of the file because the modification history information had been cleared.

I think they might have made assumptions about me they shouldn't have. Because I have a pretty unique knowledge of a lot of the file formats that Microsoft Applications use. One of those file formats being Excel. Excel files are actually a container for other files. It is a ZIP file and inside are a bunch of XML files. Inside of one of those XML files contains information about the person that created the file. Their first and last name and also the date the file was created. It also contains the first and last name as well as date and time of the last person to modify the file. As it turns out the person that created the file is not employed by the State Forest Service. They happen to be employed by the county assessors office. The other thing is the report that was supposed to have been send should have been dated near May 5th but instead it had a date of February 22. hat I was given was made before any information was available from the Forest Service and the person that made it didn't even work for the Forest Service. The last person to modify the file happens to be the guy that I asked to get the report for me. He modified the file 10 minutes before sending it to me. I suspect that was to clear the history. That was done to try and cover where the file actually originated from. But they didn't know that the information was not removed, it's simply not displayed. 

Following my discovery of fraudulent documents being sent to me I confronted the guy that works for the state. I asked why I was lied to about the source of the report and I asked why there was an attempt to hide the source of the information. I never got a response back from that guy. So yesterday I get an e-mail from the head of the Division of Property Taxation. There is no person that sits higher in this department.  I was told that from this point moving forward I am to speak with them directly.

My response was a recap of the things I had discussed with the previous person except I was being a little more abrupt about it. Little more abrasive in what I was saying because now I am repeating myself. I once again pointed out the things I found to be strange and not normal and gave examples of real data from the county and links to where that data is located... 

The e-mail I got from that person had answered a request that I made about getting a copy of the data that was used in the bias audit that the state had done in 2021. That bias audit was done by a contracted 3rd party company. Now get this, I was told that the state does not have a copy of the data so they are not able to give it to me. I was told the data from the county does not get sent to the state.... Now that seems kind of odd to me.... How is the state able to verify the accuracy of the data if it is never sent to them? Not having the ability to check the data before the bias audit is done is self defeating. It makes doing the audit almost pointless because the county would be able to send whatever it wanted to have the audit performed. In the audit report it makes no mention of checking the data to ensure it has not been tampered with. 

I also find it rather odd that the state is the one that is having the audit performed and the counties having communications with the company performing the audit almost seems like a conflict of interest.

The last paragraph I said was "I am not going to go away. I am not crazy either because other people have looked at what I am showing you now and they are in agreement that something is not right. I am the person that put in the complaint but I am not just a single person. I am making this single suggestion to you and that is to actually spend the time to read the information in this e-mail more than once if needs be. Make sure to click on the links and read the information there as well. When you do you will then realize that this complaint was made for a reason that is real. I am hoping that you come to the same conclusion as I have and that you will then do a full audit on the county. "


OH you have to hear this one. This is off the wall nuts. In the county there are 1800 ish properties that are zoned as AG. The value of the property (land only) is set at 13% of the net profits of the business.  So when reversing the value the county gave using

> value * 100 / 13 

that tells me what the net profits are. Now if you take the net profits and divide them by the size of the property (in acres) you get the net profits per acre. That wouldn't seem all that unusual, except 1208 properties of the 1800 ish are all within +- 1% of 569.00. The +- 1% is due to the county rounding the value. If they didn't round the number I am willing to bet that every single one of the 1208 properties would have exactly the same net profits per acre. 

I was seeing a pattern in the values for the AG properties but I wasn't sure if I was over analyzing it or going crazy, maybe a combination of the 2. I stopped looking for the source of the pattern. When I decided to reverse the values to get the net profits that pattern became amplified into numbers that aligned with the property size. they are using a constant dollar amount and multiplying it by the property size and the value gets set as 13% of that number. almost 70% of the AG properties are valued this way.

**Comment by Joshua-Data-Wizard** [comment]
Wow that really stinks I am sorry to hear that you are having these types of issues and the data is so hard for you to acquire.  I know where I am at the desire is to be as fair and equitable as possible so we always welcome people to dispute values so that way we can strive to be more correct. It is really disappointing to hear that you are having to go through all of this I am truly sorry.

**Comment by kdschlosser** [comment]
The single thing that really burns me is not the issue with me having to fight to get information from them. It's actually them already knowing all the information I am pointing out. This is not like a Oh wow really, how did we miss that... No No No. They have known about these problems for a long time. So it's not that I am showing them something they didn't know about.

So now the problem is not about the data. The problem is now why was it not fixed when it was originally noticed? That makes a person question if the problem is just a crappy input into the models caused by inaccurate data or is it something that was intentionally done. You mix into the entire thing me having to fight to get information from them and now what do you think is going on?? It is looking more and more like intentional manipulation of the values. For what purpose I am not sure.

I do have another question for ya. How common is it to see the same property be deeded 3-4 times in a single day? It changes hands from one company to another to another and the dollar amount for the property keeps on going up. Strange thing is sometimes the first transaction the seller ends up being the buyer on the last transfer of ownership. These properties sometimes have HUGE dollar amounts for the sales price 10's of millions of dollars, the property is a 1/10 an acre or smaller?!?!??! The property also has a county value of a couple hundred dollars. 

I just came across a bunch of land that is owned by a developer. It is all zoned as Forest Land. The requirement states contiguous 40 acres. so if it's 4 plots that are 10 acres so long as they are all connected to each other that qualifies. Of all the properties this developer owns you cannot get 40 acres contiguous from any of them. and if that isn't bad enough, in 2019 the already incredibly low amount of property taxes he was paying all got refunded??!!??!??!? It's that kind of crap that simply doesn't make any sense. The other strange thing is when you look at the properties tax history it doesn't show any credit or funds that are returned only the bill and a payment being made.

**Comment by kdschlosser** [comment]
That e-mail that I got from the head of the property taxation division at the state. I replied to that e-mail 2 days ago and the person has not responded. This is the kind of thing that happens when shady things are discovered. They stonewall ya and stop responding. The flip side is maybe they are actually investigating, but you would think that would e-mail and let me know that is what they are doing. 100% of the time they just stop responding.


This is the kind of crap that happens.

I filed a complaint with the state department that oversees vehicle emissions. I have not been able to pass emissions in my state with the Solstice and it's not a failure caused by the vehicle. All of the emissions related components are still in place and are brand new. The top end of the engine has been replaced. The car is in new condition. I am failing because of how the vehicle is being operated during the dyno test. 

Them having the throttle at 96% open and the engine load at 100% when doing an emissions test is going to cause the engine to produce an excess amount. The test is a measurement of weight of emissions over a specific distance and time. Well in a manual transmission vehicle the test can be manipulated into a failure by the person operating the vehicles during the test. If you have to travel a distance of 1 mile in 1 minute that would mean the car would need to be going 60MPH. If the car is in 5th gear when doing this the engine RPM's would be at 2000. 2000 engine RPMs means that each cylinder will detonate exactly 1000 times a minute. so on a 4 cylinder vehicle there would be 4000 detonations and each detonation would release emissions. 

Now if you take the test except the car is in 3rd gear you will be going the same speed for the same amount of time except the engine RPMs will be 5000. So now you have a total of 10000 detonations occurring. That's 2.5 times more and that means 2.5 times more emissions are produced. The test never triggers a restart because it only monitors vehicle speed not engine speed. This is what they were doing with my vehicle. My car was actually overheating when it came off the dyno because they didn't have enough air from their fan blowing into the radiator to cool the thing down. 

The complaint I made was about the vehicle not being operated properly during the test but the state kept on wanting me to bring my now unregistered vehicle an hour away so they could look at the car to see why it failed. I kept on saying to them I know why the car failed and my complaint is not about the car failing the complaint is about how the car was operated. Now get this. They must think I am some kind of an idiot. They said to me in order to determine if the vehicle was not operated properly during the test they need to inspect the car.. What?!?!?! How on earth would you know how the car was driven by looking at it??? 

They have a recording of all of the data my car sent during the entire length of the test, all they have to do is look at the data and they would be able to see it was not being operated correctly. I told them that I am not spending 1000 dollars to have the vehicle towed an hour each way for it to be inspected. If they pay for the towing they are more than welcome to inspect the vehicle. They closed the complaint stating I am unwilling to have them inspect the vehicle. 


The really messed up thing is I had to take one of my Lincoln's in for emissions so I had them test both. The Solstice went first and it failed. Now the Lincoln which is only 2 years newer that the Solstice but it has over 200K miles on it, cheap crappy Chinese e-bay non certified  catalysts on it. Check engine light is on because the timing chain needs to be replace so the engine so the timing is off. exhaust leak before one of the catalysts that is easily heard. This car should have failed big time. Same lane same operator. AWD car that has an automatic transmission. car passed. Not only did it pass, it passed with flying colors! Not even close to any of the limits, this vehicle was operated like it should have been. They were not able to manipulate the test by running the car in the wrong gear.

So you have a car with 59K miles on it 1/2 the engine is brand new, entire emissions system is brand new and all EPA certified and another that has all kinds of issues with it and non EPA certified parts on it and the latter passes and the other fails. Now does that make any sense??

**Comment by Joshua-Data-Wizard** [comment]
Wow  that all really stinks to hear and I am really sorry to hear about all of it. It is crazy how things work sometimes it is really disheartening. 

To answer your question unfortunately it isn't un common to hear about a property being deeded 3-4 times in a day. some of the reasons can be someone has plans to break it apart and once it is bought they immediately proceed with that. They could be transferring to and from other companies for liability reasons and unfortunately sometimes people buy and sell to and with from themselves and increasing amounts to make it appear values are going up sometimes this is referred to as lot trading. There are serious legal ramifications for lot trading.  I will say some times developers will trade among themselves because one has a lot the other wants for combination purposes. but hearing of that is not exactly surprising unfortunately, many times for modeling this makes it difficult to figure out which sales are good or have a related entity buying which really would not reflect an open market transaction.

**Comment by kdschlosser** [comment]
I know that the county qualifies sales but they don't explain how they go about doing this. They remove properties that are fix and flips from the time trending model. I am not sure why because both sales would be relevant because of the initial sale being priced at the condition of the home and the second sale is once again being priced taking into account the condition of the house. The house had been remodeled and the county records re gong to reflect that remodel and price adjustments are made accordingly. 

as far as the multiple deeds being done in a single day, I can understand what someone would be trying to do when they do that But the thing is the amount of money the property is being bought and sold for is way over what it should be in the first place. millions of dollars for a tenth of an acre that is off in the middle of no where? Nothing special about the property at all. I wonder if it is being done to send money through the laundry. 


The government here in Colorado is as crooked as a question mark. We have something called TABOR here, This is a tax payers bill of rights. It limits the amount the government is allowed to grow in a single year. The absolute maximum allowed is 5.5% but it can be lower based on inflation and things of that nature. Now any time they are going to increase taxes it has to be put on the ballot, but to decrease taxes they don't have to, they can just decrease them.  So this last ballot that was voted on just a couple of days ago had an item on there called proposition HH. in this bill was lowering the assessment percentage from 6.7657 to 6.7% for the next 10 years. BUT the state gets to keep funds that have been collected that excess. Right now they have to give back that money. They state that they will distribute the money to the counties to make up for the lost property taxes collected. Last year we received about 2000 in a refund from TABOR. that measly 0.0675% is not going to come close to matching the refund. This is smoke and mirrors. If it was a tax reduction it would not need to be put to a vote. Seeing as how this thing was on the ballot it is actually an increase not a decrease. The people in office here think the public is stupid. well it got voted down by a large margin. The crazy thing is there are actually people that think it is a great thing. 

What pisses me off about the TABOR thing is the county knows if it keeps the mill rates where they are they will be collecting 40% more money in property taxes. They could lower the mill rates and that would take care of the problem. This is not going to happen. instead they are going to keep them where they are at, over collect the money and then sit on it for a year. Then they will issue checks for refunds except the refunds go to everyone that lives in the county evenly. including the people that never paid the taxes in the first place. They are taking my money and giving it to someone else without my express permission. In most parts of the world that is called theft. So the 40% extra that I am paying I will likely only get about 1/2 of it back. 


This entire thing is a game they are playing. The county keeps on saying they are under staffed. They are not under staffed. It's because they are trying to get 5 days of work done in 4 days. They shrunk down to a 4 day work week in order to make budget, the funny thing is no ones paychecks ever went down. so they are being paid for 5 days but working 4 and complaining they are under staffed. Open back up the 5th day and then they would be able to get the work done!!!.

You are a numbers person. with a population of 580,000 people does 794 million dollars sound excessive to run the county for a year? I do not know what a typical county's budget is or what the per person numbers are. It seems like it is high.

**Comment by Joshua-Data-Wizard** [comment]
we actually do the same for flipped parcels. they do cause a little head ache for us because in our jurisdiction we are not allowed to know what is on the inside of a house and the only way we can find out is by closely examining deeds and scrapping listings to gather information. unfortunately since this is not completely consistent in our data yet and it is hard to quantify the value of a renovation we will include the first sale but not the second. unfortunately each jurisdiction and state handles coding  of sales slightly differently. Sometimes when there are large dollar amounts associated with a sale its because the deed include other things such as a lease on the land or access to another property. definitely  not saying that is what happening there though.

**Comment by kdschlosser** [comment]
While the assessor is not allowed inside the home the person that does the permit inspections is. That is the person that should be getting the data put into the database in order to build you a better picture of what is inside of the home. So say someone adds an addition onto the house to add another bedroom. The home has an OWTS in order the additional bedroom to be allowed there would need to be a verification of the number of bedrooms the home has. The OWTS needs to be of sufficient size in order to support the additional bedroom. Building of a deck would trigger an inspection of building easements, this would allow for counts and sizes and qualities of extra features on the property to be established. There is always a reason that can be made for an inspection to be done in order to get a building permit. The inspector would be given a sheet that the assessors office has created that would get filled out to the best of their ability and within the  scope of what they would be allowed to measure or check. As an example. If it is a kitchen remodel getting a count on the number of bathrooms would be a bit of a stretch. It can be done if the home was a ranch and the inspector had to check the plumbing. Counting the number of 3" drops from above would get you a bathroom count. 

It would sure give you a better data set to work from that's for sure. 

The other thing is it is easy to quantify the value of a remodel. The cost of a permit is based off of the cost of the project. You know the cost of a permit right? then you now know the cost of having specific types of work performed. knowing that you are able to get an average price for say a kitchen remodel or a bathroom remodel. The remodel would need to be deprecated in value as time passes and that too is easy to figure out. What is the one thing in an economy that almost never goes backwards. Inflation. The inflation amount is what the deprecation amount is. So say this year the average cost to remodel a full bath is 20 grand and next year it's 22 grand. well the bathroom remodel that took place for 20 grand would be an 18000 change in the value of the house next year. a kitchen remodel in our county triggers what they classify as a 40% remodel of a home. an adjustment of 18.10 per sqft gets made on the above grade square footage for 5 years. That is beyond crazy that replacing the counter tops and cabinets in a kitchen causes an 18.10 per sqft increase in the value of a home. Not only is this going to over value a lot of kitchen remodels it is going to under value a bunch as well. take my home for example. It has 1135 sqft above grade. so say I do a complete kitchen remodel, new floors, new cabinets and new counter tops. that means the kitchen remodel has increased the value of my home by 20,543.50. Now my kitchen is small but there is no way I could even get just the cabinets for that much unless I went to IKEA. 

Now take a home that is 4000 sqft. a kitchen remodel is going to increase the price of that home by 72,400.00. That's a low figure. and then you have the 5500 sqft multi million dollar home that has marble floors and what have you, top of the line everything. and that would increase the value by 99,550 dollars. Say my little slice of heaven is valued at 450,000 and the next property is valued at 1.2 million and the 3rd is valued at 7 million. My house ended up with an increase of 4.57% the 4000sqft home saw an increase of 6% while the super rich seen an increase of 1.4%. That is simply not even close to being right.

This is easy math. a kitchen takes up on average 12% of a home above grade area. so in my care it's 132 sqft. That is really close to being correct. I know that an average quality remodel which is a tear out and replace would run 30,000 for my size kitchen. That puts the cost at 227.00 per sqft. If we apply that to the 5500 sqft home we end up with a value change of just under 150000. That is more appropriate. and for the 4000 sqft home you se a value change of 108,960. That is closer to being correct. The cost of having work done is a direct connection with the size of the area being updated and the quality of the materials being used, not the size of the entire home.


It's funny because I have read several articles about how much should be spent on the kitchen and quite a few of the articles state 20% of the homes value. My house is valued at 650,000 the size of my kitchen is 132 sqft.  I would have to make things out of gold bars in order to spend almost 1000 a sqft on my kitchen. Apply that same dumbass logic to a 7 million dollar home that is 5500 sqft and you end up with 2121 per sqft or a 1.4 million dollar kitchen!??!?!?  Where do the authors of these articles come up with the crap they write???

They really cause problems for people when it comes to knowing what the value of work being done on their house is. A guy in my neighborhood was looking to replace a deck with a cedar one. Now I know that cedar is expensive. the deck i 8' wide by 23 feet long.  one quite after another was 60, 70 and 80 thousand dollars.  INSANE!!! The deck was only 3' off the ground!!! 10K in materials and that is stretching it. 3 to 4 days work for 3 guys.. It's like the quote I got to stick build a single car one story garage, 250,000... ummm how about NO. That's close to what I paid for my whole house.

**Comment by Joshua-Data-Wizard** [comment]
It is really interesting to hear how other areas tackle these problems. We are not quite as lucky in that the office that handles inspections of permits does not report to us at all or is even remotely connected so we have no control over what they are looking for. In addition certain areas require permits for certain things while others do not. Unfortunately in areas a person could remodel their kitchen and do this without permits and be fine while in a different area they do require the permit so that is why for indoor renovations we do not try to capture them but like you said for additions and new construction we do catch all of them! Also unfortunately for us the cost on the permit is provided by the contractor and this number has been found to be horribly inconsistent since it is self reported and many number appear to be erroneous so we can't use it :'(  The way we try to adjust for new decks and other additions and avoid the issue as mentioned above is to use the models and statistical testing to tell us the "value" added to the property. As you pointed out the cost and value are not necessarily the same thing one might spend a ton on a new patio but it might not substantially increase the value of the home in the market or potentially vice versa. Again I am not saying this is the best way to do it by any means just one way we have tried to tackle it to make sure we have statistical evidence to support the value.  There is a ton of literature out there and while some works great in areas it might not work well in others.  For what we have done, using the statistical and model approach has led to some promising results to find equitable adjustments.

**Comment by kdschlosser** [comment]
I honestly don't know how our county does it. The only way I can think of is the inspectors do it. I can tell you that a permit is needed for EVERYTHING in my county. even if you put up one of those portable garages. you know the metal frames with the tarp that gets bungee'd to the frame. a permit for one of those. a permit for sheds even the small plastic looking broom closet ones. Putting up a fence? need a a permit. building a patio using pavers/patio block? need a permit.

**Comment by Joshua-Data-Wizard** [comment]
Unfortunately from jurisdiction to jurisdiction and state to state the rules and regulations can be quite different.
